expected/13.11/extra/bool.out

```diff
@@ -0,0 +1,3652 @@
+--Testcase 53:
+EXPLAIN (VERBOSE, COSTS OFF)
+UPDATE "type_BOOLEAN" SET b = NULL WHERE NOT b;
+                                             QUERY PLAN                                             
+----------------------------------------------------------------------------------------------------
+ Update on public."type_BOOLEAN"
+   ->  Foreign Update on public."type_BOOLEAN"
+         SQLite query: UPDATE main."type_BOOLEAN" SET `b` = NULL WHERE ((NOT sqlite_fdw_bool(`b`)))
+(3 rows)
+
+--Testcase 54:
+EXPLAIN (VERBOSE, COSTS OFF)
+DELETE FROM "type_BOOLEAN" WHERE b;
+                                     QUERY PLAN                                     
+------------------------------------------------------------------------------------
+ Delete on public."type_BOOLEAN"
+   ->  Foreign Delete on public."type_BOOLEAN"
+         SQLite query: DELETE FROM main."type_BOOLEAN" WHERE (sqlite_fdw_bool(`b`))
+(3 rows)
+
+--Testcase 55:
+EXPLAIN (VERBOSE, COSTS OFF)
+DELETE FROM "type_BOOLEAN" WHERE NOT b;
+                                        QUERY PLAN                                        
+------------------------------------------------------------------------------------------
+ Delete on public."type_BOOLEAN"
+   ->  Foreign Delete on public."type_BOOLEAN"
+         SQLite query: DELETE FROM main."type_BOOLEAN" WHERE ((NOT sqlite_fdw_bool(`b`)))
+(3 rows)
+
+--Testcase 56:
+EXPLAIN (VERBOSE, COSTS OFF)
+SELECT *, NOT b nb FROM "type_BOOLEAN+" b;
+                                      QUERY PLAN                                      
+--------------------------------------------------------------------------------------
+ Foreign Scan on public."type_BOOLEAN+" b
+   Output: i, b, t, l, (NOT b)
+   SQLite query: SELECT `i`, sqlite_fdw_bool(`b`), `t`, `l` FROM main."type_BOOLEAN+"
+(3 rows)
+
+--Testcase 57:
+SELECT *, NOT b nb FROM "type_BOOLEAN+" b;
+ i  | b |    t    | l | nb 
+----+---+---------+---+----
+  1 | t | integer | 1 | f
+  2 | f | integer | 1 | t
+  3 | t | text    | 4 | f
+  4 | f | text    | 5 | t
+  5 | t | text    | 4 | f
+  6 | f | text    | 5 | t
+  7 | t | text    | 3 | f
+  8 | t | text    | 3 | f
+  9 | t | text    | 3 | f
+ 10 | f | text    | 2 | t
+ 11 | f | text    | 2 | t
+ 12 | f | text    | 2 | t
+ 13 | f | text    | 3 | t
+ 14 | f | text    | 3 | t
+ 15 | t | text    | 2 | f
+ 16 | t | text    | 2 | f
+ 17 | t | text    | 1 | f
+ 18 | t | text    | 1 | f
+ 19 | t | text    | 1 | f
+ 20 | t | text    | 1 | f
+ 21 | f | text    | 1 | t
+ 22 | f | text    | 1 | t
+ 24 | f | integer | 1 | t
+ 25 | t | integer | 1 | f
+ 26 |   | null    |   | 
+(25 rows)
+
+--Testcase 58:
+CREATE FOREIGN TABLE "type_BOOLEAN_oper"( "i" int  OPTIONS (key 'true'), i1 smallint, b1 boolean, i2 smallint, b2 boolean) SERVER sqlite_svr OPTIONS (table 'type_BOOLEAN_oper');
+--Testcase 59: see INIT.SQL with mixed affinity boolean data
+SELECT * FROM "type_BOOLEAN_oper";
+  i  | i1 | b1 | i2 | b2 
+-----+----+----+----+----
+   1 |  1 | t  |  1 | t
```

Comment:
This table is too long to verify result after adding filter. Could you limit it to about 10 rows?

Reply:
SQLite tests shows us full possible input variants. So, this table represents some main possible combinations (unfortunately not all!) of boolean data with `text` affinity. This is yet reduced test. Please refer 
```sql
CREATE TABLE "type_BOOLEAN_oper" AS
WITH booldata AS (
	SELECT row_number() over () i, column1 AS b
	  FROM (VALUES ('Yes'), ('YeS'), ('yes'), ('no'),('No'), ('nO'), ('off'),('oFf'),('on'),('ON'),('t'),('T'),('Y'),('y'),('F'),('f'),('n'),(0),(1),('0'),('1'), (NULL))
```
What is your opinion about this, @nxhai98 ? Something like `LIMIT 10` is possible, but in this case a calculations with some combinations will be untested. I have check this tables with `i1` and `i2` list for needed TCs. The table represents cartesian product of initial data.

Reply:
@nxhai98 , maybe `1::bool` firsts and than `0::bool` will be helpful for simple checking?

```sql
(VALUES
('Yes'), ('YeS'), ('yes'),('on'),('ON'),('t'),('T'),('Y'),('y'),(1),,('1'),('1'),('true'),('tRuE'),
('no'),('No'), ('nO'), ('off'),('oFf'),('F'),('f'),('n'),(0),('0'),('false'),('fALsE'),
(NULL))`
```

Reply:
I think the test for `bool literal` combinations unnecessary. For example:   
Combination `sqlite_fdw_bool('tRuE') AND sqlite_fdw_bool('false')` depends on the result of `sqlite_fdw_bool` for each bool literal and it has been tested before.    

and the main purpose of this case is bool operator pushed down.

Reply:
Thanks, I will reduce this to `DISTINCT` without row number.

Reply:
Fixed in https://github.com/pgspider/sqlite_fdw/pull/88/commits/6ccdf191564693acfe56375bde3d7a6265b65b21 to unique combinations only, please verify @nxhai98.

Reply:
Thank you. please help me resolve this thread.